pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -140,17 +141,25 @@ func validateKubeProxyConntrackConfiguration(config kubeproxyconfig.KubeProxyCon
 	return allErrs
 }
 
-func validateProxyMode(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) field.ErrorList {
+func validateProxyMode(config *kubeproxyconfig.KubeProxyConfiguration, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	switch mode {
+
+	ipvsFeatureGate := fmt.Sprintf("%s=true", features.SupportIPVSProxyMode)
+	if (strings.Contains(config.FeatureGates, ipvsFeatureGate) && config.Mode != kubeproxyconfig.ProxyModeIPVS) ||
```

Comment:
Same above, people can enable IPVS feature gateway, but it doesn't mean they always want to use IPVS proxy mode.

Reply:
>IIUC, what you say is also means that users should specify both the gate and the ipvs mode if they DO want to use IPVS mode, right?

This `if condition` ensures this, IMO. 

>Same as above, people can enable IPVS feature gateway, but it doesn't mean they always want to use IPVS proxy mode.

If users don't want to use IPVS mode, they shouldn't enable the feature gate. e.g., if user only want use iptables mode, just set the mode and leave the feature gate unset would be enough.

Reply:
I think the only condition that make sense is,

```
features.SupportIPVSProxyMode=true && proxyMode=ipvs
```

then it means start IPVS-based kube-proxy.

Reply:
Otherwise, kube-proxy will try other proxy modes.

Reply:
>Otherwise, kube-proxy will try other proxy modes.

This is an implicit behavior. That's why I (or maybe other users as well) get confused.

@kubernetes/sig-network-misc any thoughts?